src/main/scala/firrtl/transforms/ConstantPropagation.scala

```diff
@@ -160,7 +160,8 @@ class ConstantPropagation extends Transform with DependencyAPIMigration with Res
         case IntWidth(b) => b
       }
 
-      val v: Seq[Boolean] = s"%${w}s".format(a.value.toString(2)).map(_ == '1')
+      val maskedValue = a.value & ((BigInt(1) << w.toInt) - 1)
+      val v: Seq[Boolean] = s"%${w}s".format(maskedValue.toString(2)).map(_ == '1')
```

Comment:
Related to my comment https://github.com/freechipsproject/firrtl/pull/1745#discussion_r452448845 (although my suggestion would not work for this case), it seems that masking of BigInts based on width is a common theme of bugs, I think we should have a utility for it.

Reply:
I added a `maskBigInt` helper to `Utils`